### ISSUE TITLE
Errors Built with Visual Studio 2017

### ISSUE THREAD

**Issue by rathaROG** [xianyi/OpenBLAS#1344]
Hi I've tried to build the latest version with Visual Studio Enterprise 2017. The debug mode is OK, but I've found these issues in release mode. 

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB6006	"cmd.exe" exited with code 9009.	genf77blas	C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets	171	
Error	MSB6006	"cmd.exe" exited with code 9009.	gencblas	C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets	171	
```

Please help me!

**Comment by brada4** [comment]
Can you produce full output leading to error and attach it? Line in your targets file does not relate to openblas.

**Comment by rathaROG** [comment]
Yes I put in the block of codes below. There are over 2k of lines, and I just put the last 50 with errors inside. I think the problem is related to Visual Studio. Everything was done successfully without any problem when I did it with MSYS.

```
9>------ Build started: Project: gencblas, Configuration: Release x64 ------
10>------ Build started: Project: genconfig, Configuration: Release x64 ------
11>------ Build started: Project: genf77blas, Configuration: Release x64 ------
12>------ Build started: Project: openblas_utest, Configuration: Release x64 ------
9>Building Custom Rule D:/DEVLibs/OpenBlas/v0220/custom/CMakeLists.txt
11>Building Custom Rule D:/DEVLibs/OpenBlas/v0220/custom/CMakeLists.txt
10>Building Custom Rule D:/DEVLibs/OpenBlas/v0220/custom/CMakeLists.txt
12>Building Custom Rule D:/DEVLibs/OpenBlas/v0220/custom/utest/CMakeLists.txt
11>CMake does not need to re-run because D:/DEVLibs/OpenBlas/v0220/custom/build/CMakeFiles/generate.stamp is up-to-date.
11>'awk' is not recognized as an internal or external command,
11>operable program or batch file.
11>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 9009.
11>Done building project "genf77blas.vcxproj" -- FAILED.
10>CMake does not need to re-run because D:/DEVLibs/OpenBlas/v0220/custom/build/CMakeFiles/generate.stamp is up-to-date.
10>Generating openblas_config.h
12>CMake does not need to re-run because D:/DEVLibs/OpenBlas/v0220/custom/build/utest/CMakeFiles/generate.stamp is up-to-date.
12>utest_main.c
12>d:\devlibs\openblas\v0220\custom\utest\ctest.h(331): warning C4312: 'type cast': conversion from 'unsigned int' to 'void *' of greater size
12>d:\devlibs\openblas\v0220\custom\utest\ctest.h(401): warning C4312: 'type cast': conversion from 'unsigned int' to 'ctest *' of greater size
12>d:\devlibs\openblas\v0220\custom\utest\ctest.h(749): warning C4996: 'sprintf': This function or variable may be unsafe. Consider using sprintf_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
12>C:\Program Files (x86)\Windows Kits\10\Include\10.0.16299.0\ucrt\stdio.h(1772): note: see declaration of 'sprintf'
12>test_amax.c
9>CMake does not need to re-run because D:/DEVLibs/OpenBlas/v0220/custom/build/CMakeFiles/generate.stamp is up-to-date.
9>'sed' is not recognized as an internal or external command,
9>operable program or batch file.
9>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 9009.
9>Done building project "gencblas.vcxproj" -- FAILED.
12>D:\DEVLibs\OpenBlas\v0220\custom\utest\test_amax.c(36): warning C4312: 'type cast': conversion from 'unsigned int' to 'void *' of greater size
12>D:\DEVLibs\OpenBlas\v0220\custom\utest\test_amax.c(42): warning C4305: '=': truncation from 'double' to 'float'
12>D:\DEVLibs\OpenBlas\v0220\custom\utest\test_amax.c(39): warning C4305: 'initializing': truncation from 'double' to 'float'
12>Generating Code...
12>openblas_utest.vcxproj -> D:\DEVLibs\OpenBlas\v0220\custom\build\utest\openblas_utest.exe
12>Done building project "openblas_utest.vcxproj".
13>------ Build started: Project: ALL_BUILD, Configuration: Release x64 ------
13>Building Custom Rule D:/DEVLibs/OpenBlas/v0220/custom/CMakeLists.txt
13>CMake does not need to re-run because D:/DEVLibs/OpenBlas/v0220/custom/build/CMakeFiles/generate.stamp is up-to-date.
14>------ Skipped Build: Project: INSTALL, Configuration: Release x64 ------
14>Project not selected to build for this solution configuration 
========== Build: 10 succeeded, 2 failed, 0 up-to-date, 2 skipped ==========
```

**Comment by brada4** [comment]
You can get missing commands from MSYS (sed and awk are mentioned).
Maybe it is worth cleaning build directories after failed builds.

**Comment by rathaROG** [comment]
I see. What you mean is to add msys/bin to system environment variable, and then build it again. Right? Thank you so much for your help!

**Comment by theparitt** [comment]
I faced the same problem as you when I compile there are two errors using MS visual studio 2017 ( windows 10 64 -bit )  I cannot remember exactly the error code but roughly
1. cmd.exe 
2. LNK error  openblas.lib

When I checked my error log.
cmd.exe error is from the script trying to call 'perl' in the compilation script. I do not have perl on my machine. Therefore I download and install ActivePerl with set path option ( choose during installation )

Then I restart visual studio again and rebuilt ALL_BUILD. Everything is fine now.

Hope this help
